### PR TITLE
Deprecate `create` and  `create_by_shape` and replace them with `init` and  `init_by_shape`

### PR DESCRIPTION
**Pull request by MostafaDehghani**
Deprecate `create` and  `create_by_shape` and replace them with `init` and  `init_by_shape` and explicitly create a model using nn.Model given the initialized parameters as well as the module.

### CODE REVIEW

File: docs/notebooks/flax_intro.ipynb [hunk 1]
```diff
@@ -499,7 +499,8 @@
       "outputs": [],
       "source": [
         "x = jnp.ones((1, 2))\n",
-        "ys, model = Dense.partial(features=4).create(random.PRNGKey(0), x)\n",
+        "ys, initial_params = Dense.partial(features=4).init(random.PRNGKey(0), x)\n",
+        "model = nn.Model(Dense, initial_params)\n",
```
Comment (avital) [inline docs/notebooks/flax_intro.ipynb]: This isn't correct, you need to first do:

```
model_def = Dense.partial(feature=4)
```

And then use `model_def` in both the calls to `init` and the `nn.Model` constructor.

File: examples/nlp_seq/train.py [hunk 2]
```diff
@@ -106,6 +108,7 @@ def create_optimizer(model, learning_rate):
       eps=1e-9,
       weight_decay=FLAGS.weight_decay)
   optimizer = optimizer_def.create(model)
+  # TODO(jheek): remove optimizer.replicate
```
Comment (avital) [inline examples/nlp_seq/train.py]: I think this change came in by accident?

File: docs/faq.md [hunk 3]
```diff
@@ -0,0 +1,250 @@
+# The Flax FAQ
```
Comment (avital) [inline docs/faq.md]: Can you please remove the FAQ from this pull request? That should be in a separate commit.

You can do this with `git rebase -i` to remove the FAQ commit from this branch's history, then `git push --force`.

File: docs/notebooks/flax_intro.ipynb [hunk 4]
```diff
@@ -489,7 +489,7 @@
         "Model is a thin abstraction around a Module and a set of parameters.\n",
```
Comment (avital) [inline docs/notebooks/flax_intro.ipynb]: Did you update the docs manually or from within Colab? I ask because it's easy to accidentally make these not build. If you edited them manually, please follow the instructions in docs/README.md to re-build the docs and make sure that the Flax Intro still appears and looks correct.

File: docs/notebooks/flax_intro.ipynb (hunk 1 above)
Comment (MostafaDehghani) [reply docs/notebooks/flax_intro.ipynb]: Oh, yes! Done.

File: examples/nlp_seq/train.py (hunk 2 above)
Comment (MostafaDehghani) [reply examples/nlp_seq/train.py]: oh, yes! removing it...

File: docs/notebooks/flax_intro.ipynb (hunk 4 above)
Comment (MostafaDehghani) [reply docs/notebooks/flax_intro.ipynb]: I see.
I compiled the docs to make sure everything is fine.

File: docs/faq.md (hunk 3 above)
Comment (MostafaDehghani) [reply docs/faq.md]: Done!

File: examples/wip/moco/train_moco.py [hunk 5]
```diff
@@ -141,8 +141,9 @@ def create_model(key, batch_size, image_size, model_def):
   input_shape = (batch_size, image_size, image_size, 3)
   with flax.nn.stateful() as init_state:
     with flax.nn.stochastic(jax.random.PRNGKey(0)):
-      (_, _), model = model_def.create_by_shape(
+      (_, _), initial_params = model_def.initby_shape(
```
Comment (marcvanzee) [inline examples/wip/moco/train_moco.py]: Typo

File: examples/wip/moco/train_moco.py [hunk 6]
```diff
@@ -151,8 +152,9 @@ def create_linear_classifier(key, batch_size, feature_size, num_classes):
   input_shape = (batch_size, feature_size)
   model_def = flax.nn.Dense.partial(features=num_classes)
   with flax.nn.stateful():
-    _, model = model_def.create_by_shape(
+    _, initial_params = model_def.initby_shape(
```
Comment (marcvanzee) [inline examples/wip/moco/train_moco.py]: Typo

File: examples/wip/moco/train_moco.py (hunk 5 above)
Comment (MostafaDehghani) [reply examples/wip/moco/train_moco.py]: oh, sorry for that! good that you caught it.

File: examples/wip/moco/train_moco.py (hunk 6 above)
Comment (MostafaDehghani) [reply examples/wip/moco/train_moco.py]: Fixed!

**Comment by avital** [comment]
Thanks Mostafa!
